src/Microsoft.ML.TimeSeries/DTRootCauseLocalization.cs

```diff
@@ -0,0 +1,305 @@
+            }
+
+            private void CheckRootCauseInput(RootCauseLocalizationInput src, IHost host)
+            {
+                if (src.Slices.Count < 1)
+                {
+                    throw host.Except($"Length of Slices must be larger than 0");
+                }
+
+                bool containsAnomalyTimestamp = false;
+                foreach (MetricSlice slice in src.Slices)
+                {
+                    if (slice.TimeStamp.Equals(src.AnomalyTimestamp))
+                    {
+                        containsAnomalyTimestamp = true;
+                    }
+                }
+                if (!containsAnomalyTimestamp)
+                {
+                    throw host.Except($"Has no points in the given anomaly timestamp");
+                }
+            }
+
+            private void LocalizeRootCauses(RootCauseLocalizationInput src, ref RootCause dst)
+            {
+                DTRootCauseAnalyzer analyzer = new DTRootCauseAnalyzer(src, _parent._beta) ;
+                dst = analyzer.Analyze();
+            }
+        }
+    }
+
+    /// <summary>
+    /// <see cref="IEstimator{TTransformer}"/> for the <see cref="DTRootCauseLocalizationTransformer"/>.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | <xref: Microsoft.ML.Transforms.TimeSeries.RootCauseLocalizationInput> |
+    /// | Output column data type | <xref:System.Drawing.RootCause> |
+    /// | Exportable to ONNX | No |
+    ///
+    /// The resulting <xref:Microsoft.ML.Transforms.Image.DTRootCauseLocalizationTransformer> creates a new column, named as specified in the output column name parameters, and
+    /// localize the root causes which contribute most to the anomaly.
+    /// Check the See Also section for links to usage examples.
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="TimeSeriesCatalog.LocalizeRootCauseByDT" />
+    public sealed class DTRootCauseLocalizationEstimator : TrivialEstimator<DTRootCauseLocalizationTransformer>
+    {
+        internal static class Defaults
+        {
+            public const double Beta = 0.5;
+        }
+
+        /// <summary>
+        /// Localize root cause.
+        /// </summary>
+        /// <param name="env">The estimator's local <see cref="IHostEnvironment"/>.</param>
+        /// <param name="columns">The name of the columns (first item of the tuple), and the name of the resulting output column (second item of the tuple).</param>
+        /// <param name="beta">The weight for generating score in output result.</param>
+        [BestFriend]
+        internal DTRootCauseLocalizationEstimator(IHostEnvironment env, double beta = Defaults.Beta, params (string outputColumnName, string inputColumnName)[] columns)
```

Comment:
>[] [](start = 157, length = 2)

Since this transformer is non-trainable (meaning, the estimator doesn't need to look at the data in order to create a transformer), there is no advantage to supporting multiple input/output columns here - if there is ever a scenario where there are multiple input columns, the user can create an estimator chain instead.

Reply:
May I ask my question, ImageLoading and TextNormalizing are also non-trainable, however they defines this columns parameter, could you help explain when to use it and when not?

Reply:
> > [] [](start = 157, length = 2)
> 
> Since this transformer is non-trainable (meaning, the estimator doesn't need to look at the data in order to create a transformer), there is no advantage to supporting multiple input/output columns here - if there is ever a scenario where there are multiple input columns, the user can create an estimator chain instead.

I find that ImageLoading and TextNormalizing are also non-trainable, however they defines this columns parameter, could you help explain when to use it and when not?

Reply:
> > > [] [](start = 157, length = 2)
> > 
> > 
> > Since this transformer is non-trainable (meaning, the estimator doesn't need to look at the data in order to create a transformer), there is no advantage to supporting multiple input/output columns here - if there is ever a scenario where there are multiple input columns, the user can create an estimator chain instead.
> 
> I find that ImageLoading and TextNormalizing are also non-trainable, however they defines this columns parameter, could you help explain when to use it and when not?

I have updated the columns to outputColumnand  inputColumn, could you help check whether they are right?